sky/backends/cloud_vm_ray_backend.py

```diff
@@ -782,6 +783,38 @@ def _update_blocklist_on_azure_error(self, region, zones, stdout, stderr):
         else:
             self._blocked_regions.add(region.name)
 
+    def _update_blocklist_on_ibm_error(self, region, zones, stdout, stderr):
+
+        style = colorama.Style
+        stdout_splits = stdout.split('\n')
+        stderr_splits = stderr.split('\n')
+        errors = [
+            s.strip()
+            for s in stdout_splits + stderr_splits
+            if 'ERR' in s.strip() or 'PANIC' in s.strip()
+        ]
+        if not errors:
+            logger.info('====== stdout ======')
+            for s in stdout_splits:
+                print(s)
+            logger.info('====== stderr ======')
+            for s in stderr_splits:
+                print(s)
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError(
+                    'Errors occurred during launching of cluster services; '
+                    'check logs above.')
+        logger.warning('Got error(s) on IBM cluster:')
+        messages = '\n\t'.join(errors)
+        logger.warning(f'{style.DIM}\t{messages}{style.RESET_ALL}')
+
+        # add zone to blocked zones set.
+        self._blocked_zones.add(zones[0].name)
```

Comment:
Thanks, will keep conversation unresolved until merge.

Reply:
allow me to avoid diving in too deep to internal skypilot implementations: will swapping the code segment below `logger.warning(f'{style.DIM}\t{messages}{style.RESET_ALL}')` with: 
```
for zone in zones:
    self._blocked_resources.add(
        launchable_resources.copy(zone=zone.name))
``` 
will suffice?

Reply:
Yep!